xopt/generators/bayesian/bayesian_generator.py

```diff
@@ -32,44 +33,65 @@
 )
 from xopt.generators.bayesian.utils import rectilinear_domain_union, set_botorch_weights
 from xopt.numerical_optimizer import GridOptimizer, LBFGSOptimizer, NumericalOptimizer
+from xopt.pydantic import decode_torch_module
 
 logger = logging.getLogger()
 
+# It seems pydantic v2 does not auto-register models anymore
+# So one option is to have explicit unions for model subclasses
+# The other is to define a descriminated union with name as key, but that stops name field from
+# getting exported, and we want to keep it for readability
+# See https://github.com/pydantic/pydantic/discussions/5785
+
+# Update: using parent model now seems to work, keeping this just in case
+# T_ModelConstructor = Union[StandardModelConstructor, TimeDependentModelConstructor]
+# T_NumericalOptimizer = Union[LBFGSOptimizer, GridOptimizer]
+
 
 class BayesianGenerator(Generator, ABC):
     name = "base_bayesian_generator"
-    model: Model = Field(
+    model: Optional[Model] = Field(
         None, description="botorch model used by the generator to perform optimization"
     )
-    n_monte_carlo_samples = Field(
+    n_monte_carlo_samples: int = Field(
         128, description="number of monte carlo samples to use"
     )
-    turbo_controller: TurboController = Field(
+    turbo_controller: SerializeAsAny[Optional[TurboController]] = Field(
         default=None, description="turbo controller for trust-region BO"
     )
     use_cuda: bool = Field(False, description="flag to enable cuda usage if available")
-    model_constructor: ModelConstructor = Field(
+    model_constructor: SerializeAsAny[ModelConstructor] = Field(
```

Comment:
you mentioned that we need to decide on a new name for this field correct, since pydantic v2 uses ```model_``` for something? Maybe we can rename to ```gp_constructor``` instead